c/tree.c

```diff
@@ -1145,8 +1153,15 @@ LOCAL void encode_node(MMDBW_tree_s *tree, MMDBW_node_s *node,
 
     uint32_t left =
         htonl(record_value_as_number(tree, &(node->left_record), args));
+    if (0 == left) {
```

Comment:
Yes, we can tell the two types of records apart, but if the goal is to detect cycles, that doesn't help much. The aliased record can point back to a part of the tree that eventually leads back to the same alias node. The only way to find that is to do a proper cycle check by looking at the entire tree.